packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -94,15 +97,16 @@ export class FileSystemNode implements FileSystem {
             throw this.createOutOfSyncError(file, stat);
         }
         const encoding = await this.doGetEncoding(options);
-        await fs.writeFile(FileUri.fsPath(_uri), content, { encoding });
+        const encodedContent = iconv.encode(content, encoding);
+        await fs.writeFile(FileUri.fsPath(_uri), encodedContent);
         const newStat = await this.doGetStat(_uri, 1);
         if (newStat) {
             return newStat;
         }
         throw FileSystemError.FileNotFound(file.uri, 'Error occurred while writing file content.');
     }
 
-    async updateContent(file: FileStat, contentChanges: TextDocumentContentChangeEvent[], options?: { encoding?: string }): Promise<FileStat> {
+    async updateContent(file: FileStat, contentChanges: TextDocumentContentChangeEvent[], options?: { encoding?: string, overwriteEncoding?: string }): Promise<FileStat> {
```

Comment:
we need to update `filesystem` api then

I'm still not sure about `overwriteEncoding`. Could not we call `setContent` with a new encoding separately? Does it happen that one changes content and encoding at the same time?

Reply:
> we need to update filesystem api then
>
> I'm still not sure about `overwriteEncoding`. Could not we call `setContent` with a new encoding separately? Does it happen that one changes content and encoding at the same time?

@akosyakov As I see it, the initial implementation of filesystem did not pay much attention to encodings. The "overwriteEncoding" is introduced merely because I want to solve the problem with as lease modification as possible. If we want to do it in a better, and more elegant way, then the filesystem would probably need a thorough re-design.

For the latter question, I think the answer is "yes". For example, one is editing a file encoded in a strange encoding. Just when he/she is about to save the file, he/she thinks that it's probably better to save it in UTF-8. It may not common, but it does exist and need to be handled.

Reply:
Could you give high-level overview of re-design? I don't want to do it now, just want to understand what you mean and estimate impact. I think we stick with your current proposal for now.

Reply:
> Could you give high-level overview of re-design? I don't want to do it now, just want to understand what you mean and estimate impact. I think we stick with your current proposal for now.

@akosyakov I agree that we should stick to this PR. For now, maybe we can just use the "overwriteEncoding" parameter? Because it clearly indicates that we want another encoding for saving files.

Reply:
ok, let's go with it

I just wanted to understand what you mean under `the filesystem would probably need a thorough re-design.`. That something needs thorough re-design is quite a strong statement. So far we tried to align filesystem API with apis from node.js and VS Code, that devs feel familiar working with it. That's why I'm reluctant to accept `overwriteEncoding`, it is an unusual api. I wonder why we would need to step aside from standard apis and re-design it.